dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-reports-reconciliation-summary.controller/reconciliation-summary-item-mapper.ts

```diff
@@ -0,0 +1,32 @@
+import { Bank, FeeRecordEntity, UtilisationReportEntity } from '@ukef/dtfs2-common';
+import { UtilisationReportReconciliationSummaryItem } from '../../../../types/utilisation-reports';
+
+const getCountOfReconciledFeeRecords = (feeRecords: FeeRecordEntity[]): number => {
+  return feeRecords.filter((feeRecord) => feeRecord.status === 'RECONCILED').length;
+};
+
+const getCountOfDistinctFacilities = (feeRecords: FeeRecordEntity[]): number => {
+  const distinctFacilityIds = new Set(feeRecords.map((feeRecord) => feeRecord.facilityId));
+  return distinctFacilityIds.size;
+};
+
+export const mapReportToSummaryItem = (bank: Bank, report: UtilisationReportEntity): UtilisationReportReconciliationSummaryItem => {
+  const reportReceived = report.status !== 'REPORT_NOT_RECEIVED';
+  const totalFacilitiesReported = getCountOfDistinctFacilities(report.feeRecords);
+  const totalFeesReported = report.feeRecords.length;
+  const reportedFeesLeftToReconcile = totalFeesReported - getCountOfReconciledFeeRecords(report.feeRecords);
```

Comment:
It has just occurred to me that this will need to be behind the feature flag since we don't update the fee record statuses on the 'mark as' actions so I will add that in on Monday